API/Controllers/OPDSController.cs

```diff
@@ -28,14 +29,57 @@
 using Kavita.Common;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
 using Microsoft.Extensions.Logging;
 using MimeTypes;
 
 namespace API.Controllers;
 
 #nullable enable
 
+/**
+ * Middleware that checks if Opds has been enabled for this server
+ */
+[AttributeUsage(AttributeTargets.Class)]
+public class OpdsActionFilterAttribute(IUnitOfWork unitOfWork, ILocalizationService localizationService): ActionFilterAttribute
+{
+
+    public override async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+    {
+        if (!context.ActionArguments.TryGetValue("apiKey", out var apiKeyObj) ||
```

Comment:
Let's try/catch this whole thing so that in all cases, we can at least return a proper response to the client.